src/lapack_overloads.hpp

```diff
@@ -614,7 +614,30 @@ template<> inline double real(const hmat::Z_t x) {
 template<typename T> inline double real(const T x) {
   return x;
 }
+template<typename T> double imag(const T x);
+template<> inline double imag(const hmat::C_t x) {
+  return x.imag();
+}
+template<> inline double imag(const hmat::Z_t x) {
+  return x.imag();
+}
+template<typename T> inline double imag(const T x) {
+  return x;
```

Comment:
The last one, indeed

On 12/07/16 15:13, Jerome Robert wrote:
>
> In src/lapack_overloads.hpp 
> <https://github.com/jeromerobert/hmat-oss/pull/47#discussion_r70434528>:
>
> > @@ -614,7 +614,30 @@ template<> inline double real(const hmat::Z_t x) {
> >  template<typename T> inline double real(const T x) {
> >    return x;
> >  }
> > +template<typename T> double imag(const T x);
> > +template<> inline double imag(const hmat::C_t x) {
> > +  return x.imag();
> > +}
> > +template<> inline double imag(const hmat::Z_t x) {
> > +  return x.imag();
> > +}
> > +template<typename T> inline double imag(const T x) {
> > +  return x;
>
> Shouldn't it be 0 ?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/jeromerobert/hmat-oss/pull/47/files/1ba6038ef73b684782d2a797dfcd2dbfb723b2b1..0182dfa7ba6bc8403709316f86f8b97d53b60907#r70434528>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AJsflh4JPghoxm_DHQ-BzihuQSws_VDCks5qU5L1gaJpZM4JKW-u>.
>